### ISSUE TITLE
Checkstyle `6.19` Requires Java `1.8` but is not declared

### ISSUE THREAD

**Issue by mrcritical** [checkstyle/checkstyle#3283]
Even though Java 7 is no longer supported, we still have some applications using it. The recent `6.19` release caused our builds to fail since it seems to require Java 8. This was not declared in the release notes. Was this done by accident? 

```
[DEBUG] Configuring mojo org.apache.maven.plugins:maven-checkstyle-plugin:2.17:check from plugin realm ClassRealm[plugin>org.apache.maven.plugins:maven-checkstyle-plugin:2.17, parent: sun.misc.Launcher$AppClassLoader@409a44d6]
[WARNING] Error injecting: org.apache.maven.plugin.checkstyle.CheckstyleViolationCheckMojo
java.lang.UnsupportedClassVersionError: com/puppycrawl/tools/checkstyle/api/CheckstyleException : Unsupported major.minor version 52.0
```

We reverted back to `6.18` to circumvent this issue. Hopefully we can move to 8 in the near future. However I can see this being an issue for many.

**Comment by romani** [comment]
Please provide more details on that, 6.19 is jdk7 compiled.
We did swith to jdk8 only few days ago, it is not released yet, it will be 7.0 version

**Comment by mrcritical** [comment]
Our check style plugin was defined as 
```
<plugin>
         <artifactId>maven-checkstyle-plugin</artifactId>
         <version>2.17</version>
         ....
         <dependencies>
            <dependency>
              <groupId>com.puppycrawl.tools</groupId>
              <artifactId>checkstyle</artifactId>
              <version>[6.0,7)</version>
              <scope>compile</scope>
            </dependency>
          </dependencies>
         ...
</plugin>
```

But it appears someone, from our team, added snapshot support to our repositories (I just learned). Hence it pulled the `7.0-SNAPSHOT`. Of course I would have expected that Maven range to not pull `7.0`. However removing the range has fixed this for us.

Sorry for the false alarm. It didn't make sense to me either when I saw it.
